File: Cpp/807_Max_Increase_to_Keep_City_Skyline.cpp
class Solution {
public:
    int maxIncreaseKeepingSkyline(vector<vector<int>>& grid) {
        int result = 0;
        int m = grid.size();
        int n = grid[0].size();
        vector<int> row(m, 0);
        vector<int> col(n, 0);
        
        for(int i=0; i<m; ++i){
            for(int j=0; j<n; ++j){
                row[i] = max(row[i], grid[i][j]);
                col[j] = max(col[j], grid[i][j]);
            }
        }
        
        for(int i=0; i<m; ++i){
            for(int j=0; j<n; ++j){
                result += min(row[i], col[j]) - grid[i][j];
            }
        }
        
        return result;
    }
};

File: Kotlin/807_Max_Increase_to_Keep_City_Skyline.kt
class Solution {
    fun maxIncreaseKeepingSkyline(grid: Array<IntArray>): Int {
        var result = 0
        val n = grid.size
        val row = IntArray(n)
        val col = IntArray(n)
        
        for(i in 0 until n){
            for(j in 0 until n){
                row[i] = maxOf(row[i], grid[i][j])
                col[j] = maxOf(col[j], grid[i][j])
            }
        }
        
        for(i in 0 until n){
            for(j in 0 until n){
                result += minOf(row[i], col[j]) - grid[i][j]
            }
        }
        
        return result
    }
}